### ISSUE TITLE
deCONZ: Sensor entities missing and gateway ID set to 0000000000000000

### ISSUE THREAD

**Issue by SanderDelden** [home-assistant/addons#2701]
### Describe the issue you are experiencing

Hi,

I have setup deCONZ and paired my Xiaomi Aqara sensors to it. This all works well and the metrics are shown in deCONZ. However when checking in Home Assistant the entities do not show up, even after reloading the integration. After some digging I noticed that the gateway has the ID 0000000000000000  and that this could lead to issues. I then followed all steps found here: https://github.com/home-assistant/addons/blob/master/deconz/DOCS.md#my-gateway-shows-up-in-home-assistant-with-id-0000000000000000 but unfortunately to no avail.

I do have some IKEA Tradfri lights also added to deCONZ and these show up in Home Assistant just fine. It seems just the sensors are not working correctly.

Please let me know if any additional information is required.

Kind regards,
Sander

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

deCONZ

### What is the version of the add-on?

6.16.0

### Steps to reproduce the issue

1. Install deCONZ
2. Add Xiaomi Aqara sensor

## System Information

version | core-2022.10.3
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.10.5
os_name | Linux
os_version | 5.15.61-v8
arch | aarch64
timezone | Europe/Amsterdam
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.26.2
Stage | running
Available Repositories | 1186
Downloaded Repositories | 5

</details>

<details><summary>AccuWeather</summary>

can_reach_server | ok
-- | --
remaining_requests | 46

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 9.2
-- | --
update_channel | stable
supervisor_version | supervisor-2022.10.0
agent_version | 1.4.1
docker_version | 20.10.17
disk_total | 109.3 GB
disk_used | 6.2 GB
healthy | true
supported | true
board | rpi3-64
supervisor_api | ok
version_api | ok
installed_addons | InfluxDB (4.5.0), Grafana (8.0.2), SSH & Web Terminal (12.0.2), deCONZ (6.16.0)

</details>

<details><summary>Dashboards</summary>

dashboards | 2
-- | --
resources | 1
views | 5
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 6 October 2022, 13:13
-- | --
current_recorder_run | 11 October 2022, 21:03
estimated_db_size | 21.84 MiB
database_engine | sqlite
database_version | 3.38.5

</details>

<details><summary>Spotify</summary>

api_endpoint_reachable | ok
-- | --

</details>

### Anything in the add-on logs that might be useful for us?

```txt
21:19:50:793 apsUseExtPanid is 0xDDDDDDDDDDDDDDDD but should be 0, start reconfiguration
21:19:50:795 tcAddress is 0x0000000000000000 but should be 0x00212EFFFF07FA03, start reconfiguration
21:19:50:795 Skip automatic channel change, TODO warn user
21:19:58:829 Device TTL 5443 s flags: 0x3
21:20:00:792 apsUseExtPanid is 0xDDDDDDDDDDDDDDDD but should be 0, start reconfiguration
21:20:00:794 tcAddress is 0x0000000000000000 but should be 0x00212EFFFF07FA03, start reconfiguration
21:20:00:795 Skip automatic channel change, TODO warn user
21:20:10:792 apsUseExtPanid is 0xDDDDDDDDDDDDDDDD but should be 0, start reconfiguration
21:20:10:794 tcAddress is 0x0000000000000000 but should be 0x00212EFFFF07FA03, start reconfiguration
21:20:10:795 Skip automatic channel change, TODO warn user
21:20:20:793 apsUseExtPanid is 0xDDDDDDDDDDDDDDDD but should be 0, start reconfiguration
21:20:20:794 tcAddress is 0x0000000000000000 but should be 0x00212EFFFF07FA03, start reconfiguration
21:20:20:795 Skip automatic channel change, TODO warn user
21:20:30:792 apsUseExtPanid is 0xDDDDDDDDDDDDDDDD but should be 0, start reconfiguration
21:20:30:793 tcAddress is 0x0000000000000000 but should be 0x00212EFFFF07FA03, start reconfiguration
21:20:30:794 Skip automatic channel change, TODO warn user
```

### Additional information

Some more logs from HA Core which could potentially be useful:

```
2022-10-11 21:05:32.666 WARNING (MainThread) [pydeconz.utils] Received unexpected bridge id: 0000000000000000
2022-10-11 21:05:32.818 WARNING (MainThread) [pydeconz.utils] Received unexpected bridge id: 0000000000000000
2022-10-11 21:05:32.946 WARNING (MainThread) [pydeconz.utils] Received unexpected bridge id: 0000000000000000
2022-10-11 21:05:32.973 WARNING (MainThread) [pydeconz.utils] Received unexpected bridge id: 0000000000000000
2022-10-11 21:05:32.996 WARNING (MainThread) [pydeconz.utils] Received unexpected bridge id: 0000000000000000
2022-10-11 21:05:33.014 WARNING (MainThread) [pydeconz.utils] Received unexpected bridge id: 0000000000000000
2022-10-11 21:05:33.041 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up deconz platform for sensor
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 281, in _async_setup_platform
await asyncio.shield(task)
File "/usr/src/homeassistant/homeassistant/components/deconz/sensor.py", line 310, in async_setup_entry
gateway.register_platform_add_device_callback(
File "/usr/src/homeassistant/homeassistant/components/deconz/gateway.py", line 173, in register_platform_add_device_callback
async_add_device(EventType.ADDED, device_id)
File "/usr/src/homeassistant/homeassistant/components/deconz/gateway.py", line 163, in async_add_device
add_device_callback(EventType.ADDED, device_id)
File "/usr/src/homeassistant/homeassistant/components/deconz/sensor.py", line 292, in async_add_sensor
(unique_id := sensor.unique_id.rsplit("-", 1)[0])
AttributeError: 'NoneType' object has no attribute 'rsplit'
2022-10-11 21:09:55.053 WARNING (MainThread) [pydeconz.utils] Received unexpected bridge id: 0000000000000000
2022-10-11 21:09:55.089 WARNING (MainThread) [pydeconz.utils] Received unexpected bridge id: 0000000000000000
2022-10-11 21:09:55.095 WARNING (MainThread) [pydeconz.utils] Received unexpected bridge id: 0000000000000000
2022-10-11 21:09:55.102 WARNING (MainThread) [pydeconz.utils] Received unexpected bridge id: 0000000000000000
2022-10-11 21:09:55.109 WARNING (MainThread) [pydeconz.utils] Received unexpected bridge id: 0000000000000000
2022-10-11 21:09:55.118 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up deconz platform for sensor
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 281, in _async_setup_platform
await asyncio.shield(task)
File "/usr/src/homeassistant/homeassistant/components/deconz/sensor.py", line 310, in async_setup_entry
gateway.register_platform_add_device_callback(
File "/usr/src/homeassistant/homeassistant/components/deconz/gateway.py", line 173, in register_platform_add_device_callback
async_add_device(EventType.ADDED, device_id)
File "/usr/src/homeassistant/homeassistant/components/deconz/gateway.py", line 163, in async_add_device
add_device_callback(EventType.ADDED, device_id)
File "/usr/src/homeassistant/homeassistant/components/deconz/sensor.py", line 292, in async_add_sensor
(unique_id := sensor.unique_id.rsplit("-", 1)[0])
AttributeError: 'NoneType' object has no attribute 'rsplit'
```

**Comment by hendrikb** [comment]
Hey @SanderDelden - it seems to me that I have a very similar problem here. My hardware/software specs are broadly the same as yours. I use a Conbee II stick.

Everything I observe as problematic right now, used to work flawlessly a few days ago.

What I can observe:

- Aqara Devices (temperature sensors in my case) have the following problems
  - they are discovered in Deconz/Phoscon, they are  then transferred to home assistant, but with a weird name, such as "Temperature 58" (this seems pretty different from what you describe)
  - but after 1 restart of Home Assistant Core the devices are still in HA but the device entities all become "Unavailable" and are being marked with `This entity is no longer being provided by the deconz integration. If the entity is no longer in use, delete it in settings.`
  - I can continue to track & observe the respective devices in deconz/phoscon. they are still there. they work, but home assistant lost track of them.

Interestingly I noticed my problem roughly about the same time as you did. However, close to that time I have changed my zigbee setup a bit:

1. I had upgraded my conbee II stick firmware to 26780700 a few days before (and I don't dare to downgrade as of now).
2. I have connected my conbee stick to a USB extension cable, which wasn't there before (and I haven't tested the whole routine without the cable again).

Any parallels for you here, @SanderDelden?

Anybody willing to answer or contribute to @SanderDelden 's problem: feel free to do so, I don't want to highjack this topic. I'm only here to figure out whether I have the same or a similar problem (and could potentially provide more detailed input) or not.

Best,
Hendrik
